beacon-chain/sync/validate_beacon_blocks.go

```diff
@@ -456,3 +461,75 @@ func getBlockFields(b interfaces.ReadOnlySignedBeaconBlock) logrus.Fields {
 		"version":       b.Block().Version(),
 	}
 }
+
+// detectAndBroadcastEquivocation checks if the given block is an equivocating block (i.e. a different block
+// with the same slot and proposer index but different signature than one we've already seen). If an equivocation
+// is detected, it creates a proposer slashing object and broadcasts it to the network before adding it to the
+// slashing pool.
+func (s *Service) detectAndBroadcastEquivocation(ctx context.Context, blk interfaces.ReadOnlySignedBeaconBlock) error {
+	// If we've seen this proposer/slot combo before, it means this is an equivocating block
+	slot := blk.Block().Slot()
+	proposerIndex := blk.Block().ProposerIndex()
+
+	// Get the head state for block verification
+	headState, err := s.cfg.chain.HeadStateReadOnly(ctx)
+	if err != nil {
+		return err
+	}
+
+	sig1 := blk.Signature()
+	s.seenBlockLock.RLock()
+	// Create a unique cache key by combining slot and proposer index
+	// Convert slot and proposer index to 32 bytes and combine them into a single byte slice
+	b := append(bytesutil.Bytes32(uint64(slot)), bytesutil.Bytes32(uint64(proposerIndex))...)
+	existingBlock, isSeen := s.seenBlockCache.Get(string(b))
+	s.seenBlockLock.RUnlock()
+
+	if !isSeen {
+		return nil
+	}
+
+	// Type assert to get the existing block
+	existingSignedBlock, ok := existingBlock.(interfaces.ReadOnlySignedBeaconBlock)
```

Comment:
`s.seenBlockCache` in the Sync package never returns a block , we set it for example in `setSeenBlockIndexSlot`

Reply:
okay, current approach wouldn't work then. As suggested by you, made changes to use headBlock instead of blockCache for equivocation Detection.